File: julia/difference-of-squares/difference-of-squares.jl
function square_of_sum(n::Int)
    sum(1:n) ^ 2
end


function sum_of_squares(n::Int)
    #sum(map(i -> i^2, 1:n))
    sum((1:n) .^ 2)
end


function difference(n::Int)
    square_of_sum(n) - sum_of_squares(n)
end

File: kotlin/difference-of-squares/src/main/kotlin/DifferenceOfSquares.kt
import kotlin.math.pow

fun Int.pow(n: Int): Int = this.toFloat().pow(n).toInt()

class Squares(val n: Int) {

    fun sumOfSquares(): Int = (1..n).sumBy { it.pow(2) }

    fun squareOfSum(): Int = (1..n).sum().pow(2)

    fun difference(): Int = squareOfSum() - sumOfSquares()
}
